File: exercises/small_problems/easy2/sum_or_product.rb


# def calc_product(int)
#   total = 1
#   (1..int).map { |x| total *= x }
# end

puts "Please enter an integer greater than 0:"
int = gets.chomp.to_i

puts "Enter 's' to compute the sum, or 'p' to compute the product:"
operator = gets.chomp.downcase

total = case operator
        when "s"
          (1..int).sum
        when "p"
          (1..int).inject(1, :*)
        end

puts total

File: exercises/jsSmallProblems/easy1/sumOrProduct.js
const target = parseInt(prompt('Please enter an integer greater than 0:'), 0);
let operator = prompt('Enter "s" to compute the sum, '
  + 'or "p" to compute the product');

let total = 1;

operator = operator === 's' ? 'sum' : 'product';

for (let i = 2; i <= target; i += 1) {
  operator === 'sum' ? total += i : total *= i;
}

console.log(`The ${operator} of the integers `
  + `between 1 and ${target} is ${total}.`);


// FE

const sum = (accumulator, element) => accumulator + element;
const product = (accumulator, element) => accumulator * element;

const integers = [];
let arrayTotal;

const target = parseInt(prompt('Please enter an integer greater than 0:'), 0);
let operator = prompt('Enter "s" to compute the sum, '
  + 'or "p" to compute the product');

operator = operator === 's' ? 'sum' : 'product';

for (let i = 1; i <= target; i += 1) {
  integers.push(i);
}

if (operator === 'sum') {
  arrayTotal = integers.reduce(sum);
} else {
  arrayTotal = integers.reduce(product);
}

console.log(`The ${operator} of the integers `
  + `between 1 and ${target} is ${arrayTotal}.`);
